docs/sections/calculations/errors.md

```diff
@@ -565,346 +222,107 @@ $ verdi calcjob outputcat <pk_number> | less
 
 ```{code-block} console
 
-$ verdi calcjob outputls <IDENTIFIER>
+$ verdi calcjob inputcat <pk_number> | less
 
 ```
 
-and
+Again we see that, in effect, the `mickeymouse` keyword is effectively being used inside of the input file.
+Having verified this in mutiple ways, let's now correct our input parameters dictionary by leaving out the invalid key and see if our calculation succeeds:
 
-```{code-block} console
+```{code-block} ipython
 
-$ verdi calcjob outputcat <IDENTIFIER>
+In [9]: parameters_dict = {
+   ...:    "CONTROL": {
+   ...:        "calculation": "scf",
+   ...:    },
+   ...:    "SYSTEM": {
+   ...:        "ecutwfc": 30.,
+   ...:        "ecutrho": 200.,
+   ...:    },
+   ...:    "ELECTRONS": {
+   ...:        "conv_thr": 1.e-14,
+   ...:        'electron_maxstep': 3,
+   ...:    }
+   ...: }
+   ...: builder.parameters = Dict(dict=parameters_dict)
+   ...: calculation = submit(builder)
 
 ```
 
-Use the latter to verify that the energy that you have found in the last step has been extracted correctly from the output file.
-
-:::{tip}
-
-Filter the lines containing the string 'energy', e.g. using `grep`, to isolate the relevant lines.
-
-:::
-
-The results of calculations are stored in two ways: `Dict` objects are stored in the database, which makes querying them very convenient, whereas `ArrayData` objects are stored on the disk.
-The choice of what to store in `Dict` and `ArrayData` nodes is made by the parser of `pw.x` implemented in the [aiida-quantumespresso](<https://github.com/aiidateam/aiida-quantumespresso>) plugin.
-
-The `TrajectoryData` output node is a type of `ArrayData`.
-Once more, use the command `verdi data array show <IDENTIFIER>` to determine the energy obtained from calculation you ran.
-This time you will need to use the identifier of the output `TrajectoryData` of the calculation, with link name `output_trajectory`.
-As you might have realized, the difference now is that the whole series of values of the energy calculated after each ionic step are stored.
-Of course, as we simply ran an `scf` calculation, there is only one value here, but as an exercise, you can restart the calculation with:
+Use `verdi process list -a -p1` to verify that the error code is different now.
+You can check again the outputs and the reports with the tools explained in this section and try to fix it yourself before going on to the next.
 
-```{code-block} python
 
-parameters['CONTROL']['calculation'] = 'vc-relax'
+## Restarting calculations
 
-```
+It turns out that your last calculation did not converge because we stopped the self-consistency iteration cycle before it converged (by only setting a max of 3 cycles).
+In this simple case, you could just re-run the calculation from scratch with a sufficient number of iterations, but for expensive calculations (including a structural relaxation or molecular dynamics), you would like instead to restart your calculation from the previous one to save time.
```

Comment:
```suggestion
In this simple case, you could just re-run the calculation from scratch with a sufficient number of iterations, but for expensive calculations (including a structural relaxation or molecular dynamics), you would like instead to restart your calculation from the latest iteration to save time.
```

Reply:
Mmmm I find it a bit confusing  that iteration here may be referring to "the last attempted calculation" or to "the last iteration of the SCF cycle". I mean, I think both apply in this case, but I would still prefer to be clearer on which one we are intending to mention. Your intention with this modif was to change the meaning from the former (last calc) to the latter (last SCF step)?

Reply:
The issue with the original sentence what there the target/object was not clear at all. It just states "previous one", which makes no reference to what is referenced. In my opinion it's clearer to explicitly write it out and say the we are truly using the latest calculation, i.e., the latest iteration of whatever calculation cycle we were doing. (Also, I don't believe it's an "attempted" calculation, but rather an actual, complete calculation that was just stopped, because it wasn't allowed to run for more cycles.)
But feel free to change my suggestion, it should just be clear that we mean we are truly reusing the parameters and results, i.e., the output values, of the latest calculation in the iteration cycle.

Reply:
Nono, I agree with what you say about it being not clear. Probably using "iteration" is still an improvement, but I feel I could try to be even more specific. I'll try to think of another wording for this.